veps/vep-1.md

```diff
@@ -0,0 +1,90 @@
+2. Docker images will be tagged weekly as `COMPATIBILITY.yy+YYYYMMDD`.
+3. Each supported version of Vitess will be tagged weekly.
+4. Each `COMPATIBILITY` number will be supported by the Open Source Vitess community for 9 months after the initial release. _Supported Fixes to Stable Releases_ is further described below.
+
+Monotonic versioning itself is a modified version of [Semantic Versioning](https://semver.org/).
+
+## Backwards and Forwards Compatibility Promise
+
+For each `COMPATIBILITY` number, both backwards and forwards compatibility **MUST** be ensured. To use hypothetical examples:
+
+### Example 1: Deprecating an unsafe of architecturally complex feature
+
+Version `3.xx` of Vitess supports a query that is determined to be dangerous, and/or requires complex architectural support which Vitess needs to be deprecated to improve the code quality of the project:
+
+```
+# A hypothetical dangerous query!
+SELECT 1;
+```
+
+Vitess `4.xx` **MUST** still support this query, because blocking it would be backwards incompatible. However, Vitess `4.xx` **MAY** add an option to _prevent unsafe queries_. It is expected that the default behavior of _prevent unsafe queries_ would be `FALSE`. Executing the query **SHOULD** result in a deprecation warning.
+
+When Vitess `5.xx` is released, the default for _prevent unsafe_queries_ **MAY** be changed to `TRUE`. Waiting until Vitess `5.xx` to change the default makes the behavior _forwards compatible_ because users can test out the change prior to upgrading. Attempting to execute the query **SHOULD** produce an error stating that the `4.xx` behavior can be restored by changing a setting, but the setting is deprecated and will be removed soon.
+
+When Vitess `6.xx` is released, the option to _prevent unsafe queries_ **MAY** be removed with the implied behavior that the option is permanently `TRUE`.
+
+_Based on the impact of functionality being removed, the Vitess team **MAY** at their discretion extend the deprecation cycle. This example serves to demonstrate the minimum time required._
+
+### Example 2: Deprecating and removing an obsolete utility
+
+Version `3.xx` of Vitess supports a utility that has been made obsolete by recently added functionality:
+
+```
+vtdescribe: an obsolete utility
+```
+
+Vitess `4.xx` of Vitess **SHOULD** invoke a warning when attempting to use this utility, with the utility completely removed in Vitess `5.xx`.
+
+_This change is backward compatible in `Vitess 4.xx` because the utility can still be used. It is forwards compatible because attempting to use it will suggest the alternative functionality which is available._
+
+## Supported Fixes to Stable Releases
+
+Because the release includes a new release every 12 weeks, bug fixes will typically only be pushed to the master branch. An exception to this process will be made for high severity bugs, in which case a fix will be available in the weekly release for all currently supported versions.
```

Comment:
```suggestion
Because the release cycle includes a new release every 12 weeks, bug fixes will typically only be pushed to the master branch. An exception to this process will be made for high severity bugs, in which case a fix will be available in the weekly release for all currently supported versions.
```